### ISSUE TITLE
USDA database csv import [not finished]

### ISSUE THREAD

**Issue by atherdon** [GroceriStar/sd#471]
**Is your feature request related to a problem? Please describe.**
#387 

**Is it was created as task at GitHub issues? Please provide an issue number or link**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by atherdon** [comment]
@ashleycribb  will be observing this task and then we'll create a plan of what changes should be made

**Comment by atherdon** [comment]
2 files are huge, so we'll need tosplit them into chunks in order to be able to upload them at github

**Comment by atherdon** [comment]
@Edebo let's try to think about this task and maybe also discuss details about moving generator out

**Comment by atherdon** [comment]
@ashleycribb I cannot split 2 excel files. my google docs are crashing when I open them.
can you split Products, Nutrient2 into a set of files each less than 20mb and upload them here: https://github.com/ChickenKyiv/awesome-food-db-strucutures/tree/master/USDA-01-06-2019

**Comment by ashleycribb** [comment]
Sure I will take a certain amont of rows and then split them up.

**Comment by ashleycribb** [comment]
I have split the two files up ... just need priviilages so that I can upload them.

**Comment by atherdon** [comment]
https://github.com/ChickenKyiv/awesome-food-db-strucutures/issues/4

in order to apply changes to repository you need to fork it, upload files
and then create a pull request to the main repository and i'll accept them

>

**Comment by ashleycribb** [comment]
Done.

**Comment by atherdon** [comment]
perfect! merged your changes. I acutally think that this separation is
important thing. because for the begining - we actually don't need the
whole pool of data. i mean for your MVP it will be enough to have slower
number of rows

>

**Comment by atherdon** [comment]
@Edebo let's explore details about this task in the same as we did it at our other task - tell me your questions here

**Comment by Edebo** [comment]
should the same be done for the remaining csv files in https://github.com/ChickenKyiv/awesome-food-db-strucutures/tree/master/USDA-01-06-2019

**Comment by atherdon** [comment]
yes please @Edebo

**Comment by atherdon** [comment]
@Edebo while you working on this script - i'm sure you find some things/code places, that can be improved by your opinion. When you'll find someplace, that worth - put code line here or maybe add `@TODO` there, so we can catch them later, please.

One of the next goals will be improve our scripts, because i feel that we have some duplicates between this parsing script and also it will help us to have tasks for a future

**Comment by atherdon** [comment]
@Edebo I assume that we converted all database files into our tailored json format, right?

**Comment by Edebo** [comment]
Yes,we did,what is left is to review the code

**Comment by atherdon** [comment]
cool! what do you mean by code review? i cannot recall

>

**Comment by Edebo** [comment]
the todos in the codes.
then i have a question,why are we storing the json files converted into generator,why not output or sd/data since we will still move the generator as a different package

**Comment by atherdon** [comment]
Do you mean why our generated json files stored not at `src/data`? I have a
same question for that :) this is how other team members did it - but yeah
- src/data should be our main place for JSON. especially when it will be
moved out

>

**Comment by Edebo** [comment]
> Do you mean why our generated json files stored not at `src/data`? 
> yes

**Comment by atherdon** [comment]
yeah it's should be solved later. can you create a new issue and explain
with details what data should be moved? We'll assign that task to our new
members - you did some durty work before and we'll focus with you on more
complex tasks.

Can you also share with me few code links to that `todos` so I can help you
a bit ?

>

**Comment by Edebo** [comment]
https://github.com/GroceriStar/sd/blob/master/generator/csv_parser.js

**Comment by Edebo** [comment]
which do you prefer as storage for our generated json files? sd/output folder or sd/data folder

**Comment by atherdon** [comment]
it will be data folder, but it will be moved outside from src ;)

>

**Comment by atherdon** [comment]
let's start with a simple thing: create a separated method, that will shore
only
https://github.com/GroceriStar/sd/blob/master/generator/csv_parser.js#L42-L57


when you'll open a PR - i'll send what do to as next step

>

**Comment by atherdon** [comment]
second PR can be to move this function outside

https://github.com/GroceriStar/sd/blob/master/generator/csv_parser.js#L23-L38

**Comment by atherdon** [comment]
remove this duplicates: https://github.com/GroceriStar/sd/blob/master/generator/csv_parser.js#L30-L31

**Comment by Edebo** [comment]
line 30 is closing brace for the for function while line 31 is for the function,except i dont get what you mean

**Comment by atherdon** [comment]
yep, i didn't update that page. new link: https://github.com/GroceriStar/sd/blob/master/generator/csv_parser.js#L23-L25

**Comment by Edebo** [comment]
line 23-25 help to detect the remaining entries that is less than 10000,it has a purpose except if there is a way to simplify it

**Comment by atherdon** [comment]
i just see 2 similar lines and i don't like it :) make a separated method that will return a value and call it inside of `splitJsonIntoFiles`

**Comment by atherdon** [comment]
check comments from this file: https://github.com/GroceriStar/sd/blob/master/generator/projects/USFA/Derivation_Code_Description/csv_parser.js - let's improve it too

**Comment by atherdon** [comment]
@Edebo you mentioned issue about file paths. Please check this thing.
Will it help?

https://github.com/GroceriStar/sd/blob/master/babel.config.js#L25-L31

**Comment by Edebo** [comment]
i tried '~': './src/data', didnt work ,may be i dont know how to use it

**Comment by atherdon** [comment]
got it. this thing works only with transpiled files. i.e. that going to be published on npm. and while generator is not a separate thing - we have this troubles

**Comment by atherdon** [comment]
what was left? i don't think that we need to cover all todos right now at generator. or maybe we should... tell me what do you think

**Comment by Edebo** [comment]
really want to clean up codes i.e making sure things work before separating generator

**Comment by atherdon** [comment]
i'm ok with that plan. i like how files start to look similar and simple. we really reducing our code dept right now

**Comment by atherdon** [comment]
and i'm still moving us backwards, by adding new todos, but it's good for code...
![image](https://user-images.githubusercontent.com/1469198/59728009-dde75b80-9240-11e9-9b69-1e13af0d5ebc.png)

**Comment by Edebo** [comment]
no problems,

**Comment by atherdon** [comment]
@ashleycribb assuming that we pretty close to finish our preparation of this dataset - what our next step should be?

**Comment by ashleycribb** [comment]
Work on Algorithm that filters food based on diabetic needs.  We have an algorithm that filters foods based on a healthy recipe, but, not necessarily a diabetics.  I have a PDF that lays out what a diabetic should be looking for when they go to the grocery store.  We need to change the algorithm we have to that.  Also I am working with a CDE (Certified Diabetic Educator) so that we can fine tune the diabetics algorithm based on the person’s demographics (man or woman is actually treated differently with nutrition and diabetes from what she has said).  Once we have the algorithm in a good form (it will filter the dataset we have with the USDA database then we know that we can take any recipe and or pre-made food that we have the nutrients for and filter it based on an individual’s needs.  After that we can start working on an individual’s preferences.  What I am thinking is going through a list and having the person pick out what are their favorite foods. Below is a simple onboarding that I had worked on.  It’s too long and wordy, but, it’s a start.  The most important image of course is the favorite foods one, but, this was an attempt of an onboarding sequence.

[cid:image001.png@01D52A62.03ACDE40]

**Comment by atherdon** [comment]
@ashleycribb image didn't upload here

**Comment by atherdon** [comment]
ok, let's split things on smaller chunks. let's extend this part and make more details. right now it's not enough to start working on it

> Work on Algorithm that filters food based on diabetic needs.  We have an algorithm that filters foods based on a healthy recipe, but, not necessarily a diabetics.

So, that algorithm is written at R language, that you shared with me before.
First problem that i see - nobody here knows R. So how we'll update it, in order to meet your requeremtns.

I assume that your PDF will give more details, but this is what first comes in mind. What I can tell right now - it's not very easy to do. maybe we'll need to think about doing other steps

**Comment by ashleycribb** [comment]
OK. I have sent the PDF and image of the walkthrough to your gmail account.  I may have R Studio and can at least load it there to see how it works so that we can see what methods we need to create in javascript.

**Comment by atherdon** [comment]
can i upload that files here?

**Comment by ashleycribb** [comment]
Sure that’s fine.

**Comment by atherdon** [comment]
Ok, let's think that this task was completed by Adebowale.
I'll keep it open, because I want to make more changes at csv_parser later and use this task as examples for checking that everything is working fine.
